src/lib.rs

```diff
@@ -215,13 +220,24 @@ fn respond_to_request(stream: &mut TcpStream, build_dir_path: impl AsRef<Path>)
 }
 
 #[derive(Debug, StructOpt)]
-pub struct Watch {}
+pub struct Watch {
+    #[structopt(long, short = "w")]
+    watch_paths: Vec<PathBuf>,
+}
 
 impl Watch {
+    pub fn new(watch_paths: Vec<impl AsRef<Path>>) -> Self {
+        Self {
+            watch_paths: watch_paths
+                .iter()
+                .map(|path| path.as_ref().to_owned())
+                .collect(),
+        }
+    }
     pub fn execute(
         &self,
-        watch_path: impl AsRef<Path>,
         mut command: process::Command,
+        build_dir_path: impl AsRef<Path>,
```

Comment:
Oh yeah it wasn't a watch path, it was the opposite: an exclusion path.

So maybe you can generalize this by adding a field `ignore_paths` in `Watch` that would match exactly like cargo-watch:

```
    -i, --ignore <pattern>...    Ignore a glob/gitignore-style pattern
```

Try first with normal path to ignore before doing the globs. We can do the globs in another pr